src/main/scala/controllers/ApplicationController.scala

```diff
@@ -38,44 +38,32 @@ class ApplicationController @Inject()(actionHandler: ActionHandler, applications
   def editSectionForm(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
     actionHandler.gatherSectionDetails(id, sectionNumber).flatMap {
       case Some((app, af, afs, opp)) =>
-        fieldsFor(sectionNumber) match {
-          case Some(fields) =>
-            applications.getSection(id, sectionNumber).flatMap { section =>
-              val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
-              actionHandler.renderSectionForm(id, sectionNumber, section, afs.questionMap, fields, noErrors, hints)
-            }
-          case None => Future(NotFound)
+        applications.getSection(id, sectionNumber).flatMap { section =>
+          val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
+          actionHandler.renderSectionForm(id, sectionNumber, section, afs.questionMap, noErrors, hints)
         }
       case None => Future(NotFound)
     }
   }
 
   def resetAndEditSection(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
-    fieldsFor(sectionNumber) match {
-      case Some(fields) =>
-        applications.clearSectionCompletedDate(id, sectionNumber).map { _ =>
-          Redirect(controllers.routes.ApplicationController.editSectionForm(id, sectionNumber))
-        }
-      case None => Future(NotFound)
+    applications.clearSectionCompletedDate(id, sectionNumber).map { _ =>
+      Redirect(controllers.routes.ApplicationController.editSectionForm(id, sectionNumber))
     }
   }
 
   def showSectionForm(id: ApplicationId, sectionNumber: Int) = Action.async { request =>
     actionHandler.gatherSectionDetails(id, sectionNumber).flatMap {
       case Some((overview, form, afs, opp)) =>
-        fieldsFor(sectionNumber) match {
-          case Some(fields) =>
-            applications.getSection(id, sectionNumber).flatMap { section =>
-              section.flatMap(_.completedAtText) match {
-                case None =>
-                  val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
-                  actionHandler.renderSectionForm(id, sectionNumber, section, afs.questionMap, fields, noErrors, hints)
-                case _ =>
-                  Future.successful(actionHandler.redirectToPreview(id, sectionNumber))
-
-              }
-            }
-          case None => Future(NotFound)
+        applications.getSection(id, sectionNumber).flatMap { section =>
+          section.flatMap(_.completedAtText) match {
+            case None =>
+              val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
```

Comment:
Could be. It's a style thing, really. Could also be `List.empty` (which just returns `Nil` under the hood). For some reason I've always just used the no-param apply method but I'm not wedded to it. I think `List.empty` would be more explicit and easier for newcomers, and it looks like it would be faster than `List()` (which actually does some work).